addons/addon-base-raas-ui/packages/base-raas-ui/src/parts/dashboard/Dashboard.js

```diff
@@ -115,18 +126,17 @@ class Dashboard extends React.Component {
   }
 
   async getCosts() {
-    const { envNameToCostInfo, envNameToIndex } = await this.getAccumulatedEnvCost();
+    const { envIdToCostInfo, envIdToEnvMetadata, duplicateEnvNames } = await this.getAccumulatedEnvCost();
 
     const projNameToUserTotalCost = {};
-    Object.keys(envNameToCostInfo).forEach(envName => {
-      const projName = envNameToIndex[envName];
+    Object.keys(envIdToCostInfo).forEach(envId => {
+      const projName = envIdToEnvMetadata[envId].index;
       if (projNameToUserTotalCost[projName] === undefined) {
         projNameToUserTotalCost[projName] = {};
       }
-      Object.keys(envNameToCostInfo[envName].pastMonthCostByUser).forEach(user => {
-        const currentUserCost = _.get(projNameToUserTotalCost, `${projName}.${user}`, 0);
-        projNameToUserTotalCost[projName][user] =
-          currentUserCost + envNameToCostInfo[envName].pastMonthCostByUser[user];
+      Object.keys(envIdToCostInfo[envId].pastMonthCostByUser).forEach(user => {
+        const currentUserCost = _.get(projNameToUserTotalCost[projName], user, 0);
```

Comment:
Emmmm I can't figure out why the previous way `_.get(projNameToUserTotalCost, `${projName}.${user}`, 0)` was not working, do you know?

Reply:
Not sure, I didn't dive deeper and assumed the syntax was off.

Reply:
Actually I just realized it is because user contains a '.' since it is an email address

Reply:
Ah! That makes sense!